control/content/controllers/content.home.controller.js

```diff
@@ -420,25 +420,54 @@
               }
               if (validateCsv(rows)) {
                 console.log(rows);
-                Buildfire[window.DB_PROVIDER].bulkInsert(rows, TAG_NAMES.PEOPLE, function (err, data) {
-                  Buildfire.spinner.hide();
-                  $scope.$apply();
-                  if (err) {
-                    buildfire.notifications.alert('Failed to import CSV. Invalid file', function() {
+                var searchEngineInsertingFinished = false;
+                for (var i = 0; i < rows.length; i++) {
+                  var searchEngineInsert = function (counter) {
+                    buildfire.services.searchEngine.insert(
+                      {
+                        tag: TAG_NAMES.PEOPLE,
+                        title: rows[counter].fName + ' ' + rows[counter].lName,
+                        description: rows[counter].position
+                      },
+                      (err, response) => {
+                        if (err) {
+                          searchEngineInsertingFinished = true;
+                          return;
+                        }
+                        if (response && response.id) rows[counter].searchEngineDocumentId = response.id;
+                        if (counter === rows.length - 1) searchEngineInsertingFinished = true;
+                      }
+                    );
+                  }
+                  searchEngineInsert(i);
+                }
 
+                var intervalId = window.setInterval(function() {
```

Comment:
Is there a reason we are using interval for 1 sec?

Reply:
@najiatallah We are using this setInterval to check is our async logic with inserting data into search engine done. This setInterval is only used when users are inserting data via .csv. With that said, imagine they are inserting 5, 10 records in search engine. Since there is no "Bulk insert" in our sdk for search engine, we will need to hit server each time for each record. For that to finish, it will need at least one second. With that said, we are checking out every 1sec is that operation done, if yes, continue with inserting records in people plugin database.